File: testprojects/java/benchmark/accuracy-test-2013-04-17/domain/indirect/allowedfrom/AccessObjectReferenceIndirect_AsParameter.java
package domain.indirect.allowedfrom;

import domain.indirect.intermediate.BackgroundService;
import domain.indirect.intermediate.ServiceTwo;

public class AccessObjectReferenceIndirect_AsParameter {
	
	private BackgroundService bckg = new BackgroundService();
	private ServiceTwo st = new ServiceTwo();

	public AccessObjectReferenceIndirect_AsParameter(){
		String s = bckg.getServiceOneNameByArgument(st.getServiceOne());
		System.out.println(s);
	}
}

File: testresources/csharp/benchmark/Domain/Indirect/ViolatingFrom/AccessObjectReferenceIndirect_AsParameter.cs
namespace Domain.Indirect.ViolatingFrom
{
	public class AccessObjectReferenceIndirect_AsParameter
	{
        private Domain.Indirect.Intermediate.BackgroundService bckg = new Domain.Indirect.Intermediate.BackgroundService();

        private Domain.Indirect.Intermediate.ServiceTwo st = new Domain.Indirect.Intermediate.ServiceTwo();

		public AccessObjectReferenceIndirect_AsParameter()
		{
			string s = bckg.GetServiceOneNameByArgument(st.GetServiceOne());            
		}
	}
}
